File: src/my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using std::placeholders::_1;
using std::placeholders::_2;


class MyNumCounter : public rclcpp::Node 
{
public:
    MyNumCounter() : Node("number_counter")
    {
        Const = 0;
        subscriber_ = this->create_subscription<example_interfaces::msg::Int64>(
            "nombre", 10,
            std::bind(&MyNumCounter::SubscribeCallback, this, std::placeholders::_1));

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("nombre_counter", 10);

        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&MyNumCounter::NumPublisher, this));

        RCLCPP_INFO(this->get_logger(), "Let's count some numbers");

        server_ = this->create_service<example_interfaces::srv::SetBool>(
            "reset_counter",
            std::bind(&MyNumCounter::callbackReset, this, _1,_2)
            );
    }

private:
    void NumPublisher()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = Const;
        publisher_->publish(msg);
    }
    void SubscribeCallback(const example_interfaces::msg::Int64::SharedPtr msg)
    {

        Const += msg->data;
    }
    void callbackReset(const example_interfaces::srv::SetBool::Request::SharedPtr request,
                              const example_interfaces::srv::SetBool::Response::SharedPtr response){
        if (request->data == true){
            response->success = true;
            response->message = "reset";
            Const = 0;
        }
        else{
            response->success = false;
            response->message = "no reset";
        }
    }

    std::int64_t Const;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Service<example_interfaces::srv::SetBool>::SharedPtr server_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNumCounter>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from example_interfaces.srv import SetBool 
class MyNumCounter(Node): 
    def __init__(self):
        super().__init__("number_counter") 
        self.count = 0
        self.subscriber_ = self.create_subscription(Int64, "nombre",self.subscriberCallback,10)
        self.publisher_ = self.create_publisher(Int64, "nombre_count", 10)
        self.timer = self.create_timer(0.5,self.publisherCallback)
        self.get_logger().info("Time to count some numbers")
        self.server = self.create_service(SetBool, "reset_counter",self.serviceCallback)
        
    def subscriberCallback(self,msg):
        self.count += msg.data


    def publisherCallback(self):
        msg = Int64()
        msg.data = self.count
        self.publisher_.publish(msg)
    
    def serviceCallback(self,request,response):
        self.get_logger().info(str(request.data))
        if request.data:
            self.count = 0
            response.success = True
        return response

def main(args=None):
    rclpy.init(args=args)
    node = MyNumCounter() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    
    main()